File: TBroadheadPaymentSchedule.py
def calculate_discount(purchase_price, is_student, is_veteran):
    discount = 0
    if is_student:
        discount = 0.07
        if is_veteran:
            discount = 0.12
    return discount

def generate_payment_schedule(purchase_price, is_student, is_veteran):
    down_payment_rate = 0.10
    annual_interest_rate = 0.15
    monthly_payment_rate = 0.05

    discount = calculate_discount(purchase_price, is_student, is_veteran)
    discounted_price = purchase_price * (1 - discount)

    down_payment = discounted_price * down_payment_rate
    loan_amount = discounted_price - down_payment
    monthly_interest_rate = annual_interest_rate / 12

    # Calculate monthly payment
    monthly_payment = (loan_amount * monthly_interest_rate) / (1 - (1 + monthly_interest_rate) ** -12)

    # Generate payment schedule
    remaining_balance = loan_amount
    schedule = []
    for month in range(1, 13):
        interest_payment = remaining_balance * monthly_interest_rate
        principal_payment = monthly_payment - interest_payment
        remaining_balance -= principal_payment

        schedule.append((month, monthly_payment, interest_payment, principal_payment, remaining_balance))

    return schedule

def main():
    # Get user input
    purchase_price = float(input("Enter the purchase price: $"))
    is_student = input("Is the buyer a student? (yes/no): ").lower() == 'yes'
    is_veteran = False

    if is_student:
        is_veteran = input("Is the student a veteran? (yes/no): ").lower() == 'yes'

    # Generate payment schedule
    payment_schedule = generate_payment_schedule(purchase_price, is_student, is_veteran)

    # Print payment schedule
    print("\nPayment Schedule:")
    print("{:<6}{:<18}{:<11}{:<13}{:<18}".format("Month", "Monthly Payment", "Interest", "Principal", "Remaining Balance"))
    for entry in payment_schedule:
        print("{:<6}${:<17.2f}${:<10.2f}${:<12.2f}${:<17.2f}".format(*entry))

if __name__ == "__main__":
    main()

File: TBroadheadPaymentSchedule.c
#include <stdio.h>

const float downPaymentRate = 0.10;
const float annualInterestRate = 0.15;
const float monthlyPaymentRate = 0.05;
const float studentDiscount = 0.07;
const float veteranDiscount = 0.12;

void printPaymentSchedule(float purchasePrice, int isStudent, int isVeteran);

int main() {
    float purchasePrice;
    int isStudent, isVeteran;

    // Get user input
    printf("Enter the purchase price: $");
    scanf("%f", &purchasePrice);

    printf("Is the buyer a student? (1 for Yes, 0 for No): ");
    scanf("%d", &isStudent);

    if (isStudent) {
        printf("Is the student a veteran? (1 for Yes, 0 for No): ");
        scanf("%d", &isVeteran);
    } else {
        isVeteran = 0; // If not a student, not applicable
    }

    // Calculate discounted price
    float discountedPrice = purchasePrice;
    if (isStudent) {
        discountedPrice -= purchasePrice * studentDiscount;
        if (isVeteran) {
            discountedPrice -= purchasePrice * veteranDiscount;
        }
    }

    // Print payment schedule
    printPaymentSchedule(discountedPrice, isStudent, isVeteran);

    return 0;
}

void printPaymentSchedule(float purchasePrice, int isStudent, int isVeteran) {
    // Calculate down payment
    float downPayment = purchasePrice * downPaymentRate;
    float loanAmount = purchasePrice - downPayment;
    float monthlyInterestRate = annualInterestRate / 12;

    // Calculate monthly payment
    float monthlyPayment = (loanAmount * monthlyInterestRate) / (1 - pow(1 + monthlyInterestRate, -12));

    // Print payment schedule
    printf("\nPayment Schedule:\n");
    printf("Month\tMonthly Payment\tRemaining Balance\n");
    
    float remainingBalance = loanAmount;
    for (int month = 1; month <= 12; ++month) {
        remainingBalance -= monthlyPayment - remainingBalance * monthlyInterestRate;

        printf("%d\t$%.2f\t\t$%.2f\n", month, monthlyPayment, remainingBalance);
    }
}
